javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,106 @@
+const myObj = {};
+const todoArr = [];
+const myMonth = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+
+// Voice assistant function
+function getReply(command) {
+
+    if (command) {
+        const commandArr = command.split(" ");
+        command = command.toLowerCase();
+
+        // Get user name
+        if (command.startsWith("hey") || command.startsWith("hello") || command.startsWith("hi") || command.startsWith("my name")) {
+            let word = commandArr.pop();
+            if (Object.keys(myObj).length === 0) {
+                myObj.name = word;
+            }
+            return `Nice to meet you ${myObj.name}`;
+
+        } else if (command.startsWith("what") && command.endsWith("name?")) {
+            if (myObj.name) {
+                return `Your name is ${myObj.name}`;
+            } else {
+                return "You have not mentioned your name, what is your name?";
+            }
+
+        } else if (command.endsWith("my name")) {
+            word = commandArr.shift();
```

Comment:
Seems that this else-if branch is not required: this case is handled above, in the first branch